packages/interactive_media_ads/ios/interactive_media_ads/Sources/interactive_media_ads/AdsRequestProxyAPIDelegate.swift

```diff
@@ -13,20 +13,24 @@ class AdsRequestProxyAPIDelegate: PigeonApiDelegateIMAAdsRequest {
   /// The current version of the `interactive_media_ads` plugin.
   ///
   /// This must match the version in pubspec.yaml.
-  static let pluginVersion = "0.2.3+11"
+  static let pluginVersion = "0.2.3+12"
 
   func pigeonDefaultConstructor(
     pigeonApi: PigeonApiIMAAdsRequest, adTagUrl: String, adDisplayContainer: IMAAdDisplayContainer,
     contentPlayhead: IMAContentPlayhead?
   ) throws -> IMAAdsRequest {
-    // Ensure adTag can append a custom parameter.
-    assert(adTagUrl.contains("?"))
-    assert(!adTagUrl.contains("#"))
+    var modifiedURL: String?
+
+    // Add a request agent only if the adTagUrl can append a custom parameter.
+    if !adTagUrl.contains("#") && adTagUrl.contains("?") {
+      modifiedURL =
+        "\(adTagUrl)&request_agent=Flutter-IMA-\(AdsRequestProxyAPIDelegate.pluginVersion)"
+    } else {
+      modifiedURL = adTagUrl
+    }
 
-    let adTagWithRequestAgent =
-      "\(adTagUrl)&request_agent=Flutter-IMA-\(AdsRequestProxyAPIDelegate.pluginVersion)"
     return IMAAdsRequest(
-      adTagUrl: adTagWithRequestAgent, adDisplayContainer: adDisplayContainer,
+      adTagUrl: modifiedURL!, adDisplayContainer: adDisplayContainer,
```

Comment:
Rather than force-unwrapping, you could assign with a ternary, giving better type safety.

```swift
let modifiedURL = !adTagUrl.contains("#") && adTagUrl.contains("?")
  ? ...
  : adTagUrl
```